src/editors/editorsContainer/TemplateMarkdown.tsx

```diff
@@ -7,22 +9,404 @@ function TemplateMarkdown() {
   const editorValue = useAppStore((state) => state.editorValue);
   const setEditorValue = useAppStore((state) => state.setEditorValue);
   const setTemplateMarkdown = useAppStore((state) => state.setTemplateMarkdown);
-  const { value, setValue} = useUndoRedo(
+  const setMarkdownEditorCommands = useAppStore(
+    (state) => state.setMarkdownEditorCommands
+  );
+
+  const { value, setValue } = useUndoRedo(
     editorValue,
     setEditorValue,
     setTemplateMarkdown // Sync to main state and rebuild
   );
 
-  const handleChange = (value: string | undefined) => {
-    if (value !== undefined) {
-      updateEditorActivity('markdown');
-      setValue(value); // Update editor state and sync
-      setTemplateMarkdown(value); 
+  const editorRef = useRef<MonacoEditorNS.IStandaloneCodeEditor | null>(null);
+
+  useEffect(() => {
+    return () => {
+      // Clear commands when unmounting
+      setMarkdownEditorCommands(undefined);
+    };
+  }, [setMarkdownEditorCommands]);
+
+  const handleChange = (val: string | undefined) => {
+    if (val !== undefined) {
+      updateEditorActivity("markdown");
+      setValue(val); // Update editor state and sync
+      setTemplateMarkdown(val);
+    }
+  };
+
+  const applyLinePrefixToggle = (prefix: string, alternatives: string[] = []) => {
```

Comment:
This seems verbose and repetitive. Can it be refactored and moved to make this easier to maintain?